xmtp_api_grpc/src/grpc_api_helper.rs

```diff
@@ -162,9 +161,12 @@ impl XmtpApiClient for Client {
                 .await
                 .map_err(|e| Error::new(ErrorKind::SubscribeError).with(e))?
                 .into_inner(),
-        };
-
-        Ok(Subscription::start(stream).await)
+        }
+        // Discard any messages that we can't parse.
+        // TODO: consider surfacing these in a log somewhere
+        .filter_map(|r| async move { r.ok() })
+        .boxed();
```

Comment:
Another nooby question - why do we need `boxed()` here? Can we use `tonic::Streaming<Envelope>` instead of `BoxStream<'static, Envelope>`?